swap/htlc.go

```diff
@@ -111,7 +122,9 @@ func (h HtlcOutputType) String() string {
 	}
 }
 
-// NewHtlc returns a new instance.
+// NewHtlc returns a new instance. For V1 and V2 scripts, receiver and sender
+// keys are expected to be in ecdsa compressed format. For V3 scripts, keys are
+// expected to be raw private key bytes.
```

Comment:
I do not think it is good design to expect different parameter values for receiver/sender key based on the value of the `version` parmater. Footguns, footguns everywhere. 

Additionally, we're not going to have access to the raw private key bytes (this is all managed in lnd for loop) - when we derive our keys, we only get the [pubkey](https://github.com/carlaKC/loop/blob/master/loopout.go#L110). We don't want to add privkey handling to loop.